lib/broccoli/ember-app.js

```diff
@@ -679,23 +674,21 @@ class EmberApp {
     @return {Tree}
   */
   _filterAppTree() {
-    if (this._cachedFilterAppTree) {
-      return this._cachedFilterAppTree;
+    if (!this._cachedFilterAppTree) {
```

Comment:
@Turbo87 - Seems good to disable return with assignment (I agree many folks are confused by these types of statements).